library/ecp.c

```diff
@@ -3198,6 +3206,57 @@ int mbedtls_ecp_gen_keypair(mbedtls_ecp_group *grp,
 {
     return mbedtls_ecp_gen_keypair_base(grp, &grp->G, d, Q, f_rng, p_rng);
 }
+#else /* ECP_FULL */
+int mbedtls_ecp_alt_gen_keypair(mbedtls_ecp_group *grp,
```

Comment:
This is looking good, but I'm not sure the implementation belongs here, as opposed to `test_suite_pk.function` which seems to be the only place using this.

Reply:
Yes, the `mbedtls_ecp_alt_gen_keypair` function itself is only explicitly called by `test_suite_pk.function`, but it is also indirectly called from `mbedtls_ecp_gen_key` which is instead used also elsewhere such as `psa_crypto_ecp.c` and `test_suite_ecp.function`

Reply:
Ah, thanks for the info. Regarding `test_suite_ecp.function` I think that's pretty simple: we'll just skip those cases. For `psa_crypto_ecp.c` I'll have a look right now. (I had grepped for `mbedtls_ecp_gen_key()` before commenting, but had completely missed the line from `psa_crypto_ecp.c` in the results. Should have looked more carefully.)

Reply:
Just had a look, and the call in `psa_crypto_ecp.c` is guarded by `MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR` so it shouldn't be a problem in builds where `KEY_TYPE_ECC_KEY_PAIR` is accelerated, which are our target here.

Reply:
>  Regarding test_suite_ecp.function I think that's pretty simple: we'll just skip those cases

Yes, sorry for not thinking about this, but it doesn't make much sense to test the ECP module with functions that are actually outside it. I'll fix this!